include/unicore-mx/nrf/51/gpio.h

```diff
@@ -0,0 +1,203 @@
+/*
+ * This file is part of the unicore-mx project.
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef LIBUCMX_GPIO_H
+#define LIBUCMX_GPIO_H
+
+#include <unicore-mx/cm3/common.h>
+#include <unicore-mx/nrf/memorymap.h>
+
+#define GPIO			(GPIO_BASE)
+
+#define GPIO_OUT			MMIO32(GPIO_BASE + 0x504)
+#define GPIO_OUTSET			MMIO32(GPIO_BASE + 0x508)
+#define GPIO_OUTCLR			MMIO32(GPIO_BASE + 0x50C)
+
+#define GPIO_IN			MMIO32(GPIO_BASE + 0x510)
+
+#define GPIO_DIR			MMIO32(GPIO_BASE + 0x514)
+#define GPIO_DIRSET			MMIO32(GPIO_BASE + 0x518)
+#define GPIO_DIRCLR			MMIO32(GPIO_BASE + 0x51C)
+
+#define GPIO_PIN_CNF(N)			MMIO32(GPIO_BASE + 0x700 + 0x4 * (N))
+
+#define PIN0			(1 << 0)
```

Comment:
I'd rather use `GPIOPIN0`...`GPIOPIN31` to avoid namespace clashes, and to make it more similar to NXP LPC gpio port